### PR TITLE
use PutUntracked to insert entries into non-unique indexes

### PR DESCRIPTION
**Pull request by jsteemann**
this will turn of any conflict checking with concurrent transactions.
however, as the keys always contain an auto-generated LocalDocumentId,
no conflicts shall even happen on the RocksDB level for these key.

this is supposed to speed up secondary index inserts and updates

### CODE REVIEW

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2956/

File: arangod/RocksDBEngine/RocksDBKey.cpp [hunk 1]
```diff
@@ -32,6 +32,29 @@ using namespace arangodb;
 using namespace arangodb::rocksutils;
 
 const char RocksDBKey::_stringSeparator = '\0';
+  
+/// @brief verify that a key actually contains the given local document id
+bool RocksDBKey::containsLocalDocumentId(LocalDocumentId const& documentId) const {
+  switch (_type) {
+    case RocksDBEntryType::EdgeIndexValue:
```
Comment (graetzer) [inline arangod/RocksDBEngine/RocksDBKey.cpp]: technically you could also add this assertion to the normal insert method

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2959/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2961/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2962/

**Pull request merged**